api/dataset.go

```diff
@@ -36,7 +34,7 @@ func (api *DatasetAPI) getDatasets(w http.ResponseWriter, r *http.Request) {
 
 	var bytes []byte
 
-	if r.Header.Get(internalToken) == api.internalToken {
+	if api.EnablePrePublishView && r.Header.Get(internalToken) == api.internalToken {
```

Comment:
I dont know if its worth splitting these 2 checks out - we can log specifically why the request was denied then - again makes figuring out whats happening from the logs much clearer.

Reply:
Done